main.go

```diff
@@ -0,0 +1,323 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"github.com/himanshuo/gameoflife/models"
+	_ "github.com/mattn/go-sqlite3"
+	"html/template"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+//_ is in order to import a package solely for its side-effects at initialization.
+//In this case, go-sqlite3's side effects are allowing sqlite3 to be usable as a
+//database for  sql.Open
+
+//the database resource
+var db *sql.DB
+var Router *mux.Router
+
+//run at start of program.
+func init() {
+
+	startDB()
+	Router = mux.NewRouter()
+
+}
+
+func startDB() error {
+	var err error
+	//sqlite 3 database is stored in /data/data.db file
+	db, err = sql.Open("sqlite3", "./data/data.db")
+	if err != nil {
+		return err
+	}
+
+	//make sure we can actually query the database.
+	if err := db.Ping(); err != nil {
+		return err
+	}
+
+	//ASSUMPTION: table exists in database
+
+	//statement creates Task table which only has a PK id and name textfields
+	sqlStmt := "create table Task (id integer not null primary key autoincrement, name text);"
```

Comment:
change to 

CREATE TABLE Task (id INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, name TEXT);"